FreeRTOS_IP_Utils.c

```diff
@@ -1170,7 +1180,14 @@ size_t FreeRTOS_min_size_t( size_t a,
 uint32_t FreeRTOS_round_up( uint32_t a,
                             uint32_t d )
 {
-    return d * ( ( a + d - 1U ) / d );
+    uint32_t ulResult = 0;
+
+    if( d != 0 )
```

Comment:
if d is zero
zero will be returned, which is wrong 
is that the intended behavior?

Reply:
I have added an assert to make sure that `d == 0` is caught in development phase.
However, since we cannot return an error with this function signature, I am returning `a` in round_up and `0` in round_down when assert is not defined.